drivers/hwinfo/hwinfo_sam.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2019 Aurelien Jarno
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <hwinfo.h>
+#include <init.h>
+#include <soc.h>
+#include <string.h>
+
+static u8_t sam_uid[16];
```

Comment:
Is this automatically protected from userspace write?

Reply:
When userspace is enabled, this is not readable or writeable from user threads. When it's not enabled, if a thread can get a pointer to it, it can access it in read/write mode. But that's not different from other data.

Reply:
I think it's ok then.